lib/datadog/tracing/contrib/karafka/events/worker/process.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require_relative '../../ext'
+require_relative '../../event'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Karafka
+        module Events
+          module Worker
+            module Process
+              include Karafka::Event
+
+              def self.subscribe!
+                ::Karafka.monitor.subscribe 'worker.process' do |event|
```

Comment:
I would not recommend using this layer for instrumentation of that type and would advise you to reconsider something "closer" to the actual execution of work.

Reply:
Would it be possible for you to explain what "closer" means here :)? I utilised what you did on instrumentation/vendors/datadog/logger_listener.rb here, but what I'm reconsidering here that we don't actually need to subscribe to this event anymore, because our goal is to have message traces for each message inside messages enumerator, so we can link them to the distributed traces. So I can remove this even if everything is getting our of control here.